### DISCUSSION TITLE
Cannot build from source on Ubuntu 24 when using an installed ffmpeg build

### DISCUSSION THREAD

**Discussion by JoeSchiff** [PyAV-Org/PyAV#1696] [4. Bugs] [answered]
## Overview

Running the instructions to build from source causes a hwaccel error when compiling against my own build of FFmpeg:
```
git clone https://github.com/PyAV-Org/PyAV.git
cd PyAV
source scripts/activate.sh
make
```
However, there is no error if I build ffmpeg:
```
git clone https://github.com/PyAV-Org/PyAV.git
cd PyAV
source scripts/activate.sh
./scripts/build-deps
make
```

## Actual behavior
Output with error:
https://gist.github.com/JoeSchiff/e4e64621a3955ec38411182f3dedcbbc

## Reproduction
```
git clone https://github.com/PyAV-Org/PyAV.git
cd PyAV
source scripts/activate.sh
make
```

## Versions

- OS: Ubuntu 24.04.1
```
$ uname -a
Linux joepers-pc 6.8.0-49-generic #49-Ubuntu SMP PREEMPT_DYNAMIC Mon Nov  4 02:06:24 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```
GPU:  Radeon RX 7600

## Additional context

ffmpeg was installed using `sudo apt install ffmpeg`

```
$ ffmpeg -version
ffmpeg version 6.1.1-3ubuntu5 Copyright (c) 2000-2023 the FFmpeg developers
built with gcc 13 (Ubuntu 13.2.0-23ubuntu3)
configuration: --prefix=/usr --extra-version=3ubuntu5 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --disable-omx --enable-gnutls --enable-libaom --enable-libass --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libharfbuzz --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-openal --enable-opencl --enable-opengl --disable-sndio --enable-libvpl --disable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-ladspa --enable-libbluray --enable-libjack --enable-libpulse --enable-librabbitmq --enable-librist --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libx264 --enable-libzmq --enable-libzvbi --enable-lv2 --enable-sdl2 --enable-libplacebo --enable-librav1e --enable-pocketsphinx --enable-librsvg --enable-libjxl --enable-shared
libavutil      58. 29.100 / 58. 29.100
libavcodec     60. 31.102 / 60. 31.102
libavformat    60. 16.100 / 60. 16.100
libavdevice    60.  3.100 / 60.  3.100
libavfilter     9. 12.100 /  9. 12.100
libswscale      7.  5.100 /  7.  5.100
libswresample   4. 12.100 /  4. 12.100
libpostproc    57.  3.100 / 57.  3.100
```

**Comment by WyattBlue** [comment]
You either need to use Ubuntu's pyav library, or both build PyAV and ffmpeg from source.

**Comment by zak-45** [comment, marked as answer]
Got the same problem. Supposed that ffmpeg version 6.1.1 should be incompatible with pyAV 14.1.

Solved it by:

Upgrade to ffmpeg 7
> sudo add-apt-repository ppa:ubuntuhandbook1/ffmpeg7
sudo apt update
sudo apt install ffmpeg

To verify:
> ffmpeg -version

this should provide:
> ffmpeg version 7.1 Copyright (c) 2000-2024 the FFmpeg developers

So ... I had been able to run:
> pip install av --no-binary av

Provide this output:
> Successfully built av
Installing collected packages: av
Successfully installed av-14.1.0


Important Note:

Updating FFmpeg via a PPA may affect other applications that depend on it. Ensure that this update won't cause compatibility issues with your existing software. Use PPAs at your own risk.

**Reply by onkelbeh** [reply]
Yep, this helped on Gentoo, too. Against ffmpeg-6.1.2 I ran into the same error. It compiled after upgrading to ffmpeg-7.1.
Same behavior: av-13.0.1 merged as well as av-14.1.0. I am using it with Home Assistant.
